File: chapter03/akkademaid-java/src/main/java/com/lp/akka/notes/ParsingActor.java
package com.lp.akka.notes;


import akka.actor.AbstractActor;
import akka.japi.pf.ReceiveBuilder;
import com.lp.akka.notes.pojo.ArticleBody;
import com.lp.akka.notes.pojo.ParseHtmlArticle;
import de.l3s.boilerpipe.extractors.ArticleExtractor;
import scala.PartialFunction;

/**
 * @author lipan
 */
public class ParsingActor extends AbstractActor {
    @Override
    public PartialFunction receive() {
        return ReceiveBuilder.
                match(ParseHtmlArticle.class, msg -> {
                    String body = ArticleExtractor.INSTANCE.getText(msg.htmlString);
                    sender().tell(new ArticleBody(msg.uri, body), self());
                }).build();
    }
}

File: chapter03/akkademaid-scala/src/main/scala/com/lp/akka/notes/ParsingActor.scala
package com.lp.akka.notes

import akka.actor.Actor
import com.lp.akka.notes.pojo.{ArticleBody, ParseHtmlArticle}

class ParsingActor extends Actor{
  override def receive: Receive = {
    case ParseHtmlArticle(key, html) =>
      sender() ! ArticleBody(key, de.l3s.boilerpipe.extractors.ArticleExtractor.INSTANCE.getText(html))
    case x =>
      println("unknown message " + x.getClass)
  }
}
